test/common/index.js

```diff
@@ -120,6 +120,18 @@ const isFreeBSD = process.platform === 'freebsd';
 const isOpenBSD = process.platform === 'openbsd';
 const isLinux = process.platform === 'linux';
 const isOSX = process.platform === 'darwin';
+const isPi = (() => {
+  try {
+    // Normal Raspberry Pi detection is to find the `Raspberry Pi` string in
+    // the contents of `/sys/firmware/devicetree/base/model` but that doesn't
+    // work inside a container. Match the chipset model number instead.
+    const cpuinfo = fs.readFileSync('/proc/cpuinfo', { encoding: 'utf8' });
+    return /^Hardware\s*:\s*(.*)$/im.exec(cpuinfo)?.[1] === 'BCM2835';
```

Comment:
Note: `BCM2835` matches what `/proc/cpuinfo` returns on the Pi 2's and 3's in our CI.
e.g. 
```
$ docker exec node-ci-stretch cat /proc/cpuinfo
processor       : 0
model name      : ARMv7 Processor rev 4 (v7l)
BogoMIPS        : 76.80
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

processor       : 1
model name      : ARMv7 Processor rev 4 (v7l)
BogoMIPS        : 76.80
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

processor       : 2
model name      : ARMv7 Processor rev 4 (v7l)
BogoMIPS        : 76.80
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

processor       : 3
model name      : ARMv7 Processor rev 4 (v7l)
BogoMIPS        : 76.80
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 4

Hardware        : BCM2835
Revision        : a22082
Serial          : 000000001acb5e3f
Model           : Raspberry Pi 3 Model B Rev 1.2
```
I haven't matched against other Pi chipsets -- I don't know, for example, whether the Pi 4's are fast enough to run these slow tests (we don't have any in the Node.js CI).

Reply:
Also `os.cpus()` doesn't return the `Hardware` field:
```
$ node -p "os.cpus()"
[
  {
    model: 'ARMv7 Processor rev 4 (v7l)',
    speed: 1200,
    times: { user: 5251030, nice: 0, sys: 634800, idle: 56307240, irq: 0 }
  },
  {
    model: 'ARMv7 Processor rev 4 (v7l)',
    speed: 1200,
    times: { user: 5691040, nice: 0, sys: 1000630, idle: 56166190, irq: 0 }
  },
  {
    model: 'ARMv7 Processor rev 4 (v7l)',
    speed: 1200,
    times: { user: 5880350, nice: 10, sys: 953020, idle: 55968650, irq: 0 }
  },
  {
    model: 'ARMv7 Processor rev 4 (v7l)',
    speed: 1200,
    times: { user: 5741270, nice: 0, sys: 945350, idle: 56215820, irq: 0 }
  }
]
```

Reply:
What is the problem with /sys? It's not available at all or that specific path doesn't exist or ?

Reply:
@mscdex That specific path doesn't exist.
```
pi@test-requireio--williamkapke-debian10-arm64--pi3-3:~ $ docker exec node-ci-stretch ls -al /sys
total 4
dr-xr-xr-x  12 root root    0 Apr  7 01:22 .
drwxr-xr-x  20 root root 4096 Apr  7 01:21 ..
drwxr-xr-x   2 root root    0 Apr  8 01:22 block
drwxr-xr-x  26 root root    0 Apr  8 01:22 bus
drwxr-xr-x  54 root root    0 Apr  8 01:22 class
drwxr-xr-x   4 root root    0 Apr  8 01:22 dev
drwxr-xr-x  11 root root    0 Apr  8 01:16 devices
drwxrwxrwt   2 root root   40 Apr  7 01:22 firmware
drwxr-xr-x   7 root root    0 Apr  7 01:22 fs
drwxr-xr-x  12 root root    0 Apr  8 01:22 kernel
drwxr-xr-x 124 root root    0 Apr  8 01:22 module
drwxr-xr-x   2 root root    0 Apr  8 01:22 power
pi@test-requireio--williamkapke-debian10-arm64--pi3-3:~ $ docker exec node-ci-stretch ls -al /sys/firmware
total 0
drwxrwxrwt  2 root root 40 Apr  7 01:22 .
dr-xr-xr-x 12 root root  0 Apr  7 01:22 ..
pi@test-requireio--williamkapke-debian10-arm64--pi3-3:~ $
```

Reply:
FWIW there should also be /proc/device-tree/model. Not sure if that's better or worse than parsing /proc/cpuinfo

Reply:
`/proc/device-tree` appears to be a symlink inside the container pointing to a non-existent location:
```
pi@test-requireio--williamkapke-debian10-arm64--pi3-3:~ $ docker exec node-ci-stretch ls -al /proc/device-tree
lrwxrwxrwx 1 root root 29 Apr  8 01:37 /proc/device-tree -> /sys/firmware/devicetree/base
pi@test-requireio--williamkapke-debian10-arm64--pi3-3:~ $ docker exec node-ci-stretch ls -al /proc/device-tree/
ls: cannot access '/proc/device-tree/': No such file or directory
pi@test-requireio--williamkapke-debian10-arm64--pi3-3:~ $
```

Reply:
See also https://github.com/moby/moby/issues/43419.

Reply:
This won't stop it running on a pi4, but as you said that doesn't matter as we don't have any:
```
Hardware	: BCM2711
Revision	: d03114
Serial		: 10000000302bfe33
Model		: Raspberry Pi 4 Model B Rev 1.4
```
Similarly the ODROID distributions give their own custom output e.g.
```
Hardware	: Hardkernel ODROID-XU4
Revision	: 0100
Serial		: 0000000000000000
```
(FYI I get timeouts in a couple of keepalive tests on that machine which is one of my own dev boxes, but no other timeouts based on the one run I've just done)